### ISSUE TITLE
Flux e2e tests: let to transformation complete even on error.

### ISSUE THREAD

**Issue by affo** [influxdata/influxdb#13975]
In `dynamic_query.flux` we have (wrapping up the script):

```
csv.from(...) |> to(bucket: BBB)
from(bucket: BBB) |> ... |> tableFind(fn: (key) => key._measurement == "CPU")
```

`tableFind` triggers an error if it doesn't find a table that satisfies `fn`. The problem is that if `from` and `to` run in parallel, then `tableFind` will error and block `to` from persisting values to InfluxDB.

The solution is to clearly split the `to` and `from` "phases" in e2e tests.

**Comment by affo** [comment]
This should be fixed once we get https://github.com/influxdata/flux/issues/1799 in and e2e test running strategy in InfluxDB is fixed accordingly.

**Comment by affo** [comment]
This should still be relevant

**Comment by wolffcm** [comment]
My understanding of this issue is:
- The e2e tests execute a `to` to populate the database with data, and also `from` to read the data written by `to`
- The Flux execution engine doesn't know that after `to` executes, it should wait until data can be read before executing the `from`
- We get around this by just executing the test script twice.  On the second execution, the data is usually (always?) there, so the test passes.
- If there is a call to `tableFind`, we try to read the data before it's there, and throw an error.  Because the throw an error, the whole script fails, and the data is never written.  Repeated executions don't fix the problem in that case.

Seems like the best fix for this would be to teach the execution engine to understand dependencies between executions and allow some operations to block when needed to preserve semantics.

**Comment by affo** [comment]
@wolffcm yeah, I agree. Issue https://github.com/influxdata/flux/issues/1799 contains informations on how the interpreter could understand those dependencies.

That issue would make it so

```
... |> to()
from() |> ....
```

Would execute serially, thanks to a general strategy of executing things when they contain side effects.

**Comment by yzhang1991** [comment]
@affo Can I take influxdata/flux#1799 as a replacement of this issue and close this?

**Comment by affo** [comment]
Yeah, I think so.

This one was specifically made to remember update how e2e tests are executed in InfluxDB once  influxdata/flux#1799 had been fixed 👍
